tiledb/api/src/array/domain.rs

```diff
@@ -106,6 +181,27 @@ impl<'ctx> Debug for Domain<'ctx> {
     }
 }
 
+impl<'c1, 'c2> PartialEq<Domain<'c2>> for Domain<'c1> {
```

Comment:
There is a proptest for reflexivity but no test that changing fields changes the result.  I need to add that.